api/client/service/update.go

```diff
@@ -358,12 +358,23 @@ func updateMounts(flags *pflag.FlagSet, mounts *[]swarm.Mount) {
 }
 
 func updatePorts(flags *pflag.FlagSet, portConfig *[]swarm.PortConfig) {
+	portSet := make(map[swarm.PortConfig]struct{})
+	for _, entry := range *portConfig {
```

Comment:
Wait: build an index by the left port (where `left:right`). That is what needs to be unique here.